piperider_cli/recipes/__init__.py

```diff
@@ -115,6 +120,32 @@ def __dict__(self):
             d['cloud'] = self.cloud.__dict__()
         return d
 
+    def is_file_specified(self):
+        return hasattr(self, 'file')
```

Comment:
It will not working as you want in this case:

```yaml
base:
  file: "{{ env_var('TYPEO_BASE_FILE') }}"
```

* There is no `TYPEO_BASE_FILE` variable, but piperider gets a file with value `"None"`
* It should be fast-fail when the file is not existing

Reply:
Got it, will add fail-fast checks.
thanks!